backend/infrahub/core/account.py

```diff
@@ -71,24 +86,104 @@ def get_permissions(self) -> list[GlobalPermission]:
             permissions.append(
                 GlobalPermission(
                     id=result.get("global_permission").get("uuid"),
-                    name=result.get("global_permission_action").get("value"),
+                    name=result.get("global_permission_name").get("value"),
                     action=result.get("global_permission_action").get("value"),
                 )
             )
 
         return permissions
 
 
-async def fetch_permissions(
-    account_id: str, db: InfrahubDatabase, branch: Optional[Union[Branch, str]] = None
-) -> dict[str, list[GlobalPermission]]:
+class AccountObjectPermissionQuery(Query):
+    name: str = "account_object_permissions"
+
+    def __init__(self, account_id: str, **kwargs: Any):
+        self.account_id = account_id
+        super().__init__(**kwargs)
+
+    async def query_init(self, db: InfrahubDatabase, **kwargs: Any) -> None:
+        self.params["account_id"] = self.account_id
+
+        branch_filter, branch_params = self.branch.get_query_filter_path(
+            at=self.at.to_string(), branch_agnostic=self.branch_agnostic
+        )
+        self.params.update(branch_params)
+
+        query = """
+        MATCH (account:CoreGenericAccount)
+        WHERE account.uuid = $account_id
+        CALL {
+            WITH account
+            MATCH (account)-[r:IS_PART_OF]-(root:Root)
+            WHERE %(branch_filter)s
+            RETURN account as account1, r as r1
+            ORDER BY r.branch_level DESC, r.from DESC
+            LIMIT 1
+        }
+        WITH account, r1 as r
+        WHERE r.status = "active"
+        WITH account
+        MATCH group_path = (account)-[]->(:Relationship {name: "group_member"})
```

Comment:
Thanks, yes you are correct. I had modified the permissions back and forth and that part seemed to work fine however I didn't try to delete anything and when I do the query breaks apart.

I did a quick test and changed the .get_query_filter_path() method to include this:

```python
        filters = []
        for idx, (branch_name, time_to_query) in enumerate(branches_times.items()):
            filters.append(
                f"(r.branch IN $branch{idx} AND r.from <= $time{idx} AND r.to IS NULL AND r.status = 'active')"
            )
            filters.append(
                f"(r.branch IN $branch{idx} AND r.from <= $time{idx} AND r.to >= $time{idx} AND r.status = 'active')"
            )
```

I.e add the part with `r.status = "active"`. While I guess we don't want this everywhere. To me it seems that the error is that when deleting a node we don't set the r.to value on those?

If I look at another query: 

```cypher
class AttributeRegexUpdateValidatorQuery(AttributeSchemaValidatorQuery):
    name: str = "attribute_constraints_regex_validator"

    async def query_init(self, db: InfrahubDatabase, **kwargs: dict[str, Any]) -> None:
        branch_filter, branch_params = self.branch.get_query_filter_path(at=self.at.to_string())
        self.params.update(branch_params)

        self.params["node_kind"] = self.node_schema.kind
        self.params["attr_name"] = self.attribute_schema.name
        self.params["attr_value_regex"] = self.attribute_schema.regex
        self.params["null_value"] = NULL_VALUE
        query = """
        MATCH p = (n:Node)
        WHERE $node_kind IN LABELS(n)
        CALL {
            WITH n
            MATCH path = (root:Root)<-[rr:IS_PART_OF]-(n)-[ra:HAS_ATTRIBUTE]-(:Attribute { name: $attr_name } )-[rv:HAS_VALUE]-(av:AttributeValue)
            WHERE all(
                r in relationships(path)
                WHERE %(branch_filter)s
            )
            RETURN path as full_path, n as node, rv as value_relationship, av.value as attribute_value
            ORDER BY rv.branch_level DESC, ra.branch_level DESC, rr.branch_level DESC, rv.from DESC, ra.from DESC, rr.from DESC
            LIMIT 1
        }
        WITH full_path, node, attribute_value, value_relationship
        WITH full_path, node, attribute_value, value_relationship
        WHERE all(r in relationships(full_path) WHERE r.status = "active")
        AND attribute_value <> $null_value
        AND NOT attribute_value =~ $attr_value_regex
        """ % {"branch_filter": branch_filter}
````
This seems a bit off to me. I'd suggest that we add a `status: str | None = None` parameter to .get_query_filter_path() so that we can inject a status there directly if there's no reason against that? Though I'm still wondering about that `r.to` field..

Reply:
if these paths need to be branch-aware, then the query will need to be updated to get the latest version of each path (preferring the path on the branch to the path on `main`) and filter to only the non-deleted ones. This is basically what the `CALL` subquery for getting the account at the beginning of this query does. If all of these paths are for branch-agnostic nodes and attributes, then I think the query will work as is